### PR TITLE
Fixes an issue where passing in customProperties would render invalidly encoded object

### PR DESCRIPTION
**Pull request by sdglhm**
## Description

This fixes an issue where when passing in customProperties to a Choice would lead to render `[object Object]` instead of properly rendering JSON structure.

## Types of changes

- [ ] Chore (tooling change or documentation change)
- [ ] Refactor (non-breaking change which maintains existing functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] My code follows the code style of this project.
- [x] I have added new tests for the bug I fixed/the new feature I added.
- [ ] I have modified existing tests for the bug I fixed/the new feature I added.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/scripts/templates.ts [hunk 1]
```diff
@@ -321,7 +321,7 @@ const templates = {
     const opt = new Option(label, value, false, active);
 
     if (customProperties) {
```
Comment (Xon) [inline src/scripts/templates.ts:323]: FYI, but this isn't a sufficient test for if a javascript object is empty

Comment (sdglhm) [reply src/scripts/templates.ts:323]: Thank you for the feedback. I'll fix the tests and update the PR.

Comment (sdglhm) [reply src/scripts/templates.ts:323]: @Xon I've updated the tests. Could you please let me know if this is correct. I'm not really great with JS as I majorly work on backends. Thank you.

**Comment by Xon** [comment]
I've implemented a more comprehensive fix as part of #1166

**Pull request closed without merging**